tests/h/search/query_test.py

```diff
@@ -505,11 +505,32 @@ def test_matches_replies_of_replies_to_an_annotation(self, Annotation, search):
 
 
 class TestTagsAggregation(object):
+    def test_it_aggregates_annotations_on_tags(self, Annotation, search):
+        for i in range(2):
+            Annotation(tags=['tag_a'])
+        Annotation(tags=['tag_b'])
 
-    @pytest.fixture
-    def search(self, search):
         search.append_aggregation(query.TagsAggregation())
-        return search
+        result = search.run({})
+
+        count_for_tag_a = next(r for r in result.aggregations.get('tags') if r["tag"] == "tag_a")['count']
+        count_for_tag_b = next(r for r in result.aggregations.get('tags') if r["tag"] == "tag_b")['count']
+
+        assert len(result.aggregations.get('tags')) is 2
+        assert count_for_tag_a is 2
+        assert count_for_tag_b is 1
+
+    def test_it_limits_annotation_aggregation_on_tags(self, Annotation, search):
+        for i in range(3):
+            Annotation(tags=['tag_a'])
+
+        search.append_aggregation(query.TagsAggregation(2))
+        result = search.run({})
+
+        count_for_tag_a = next(r for r in result.aggregations.get('tags') if r["tag"] == "tag_a")['count']
+
+        assert len(result.aggregations.get('tags')) is 1
+        assert count_for_tag_a == 3
```

Comment:
Shouldn't this count returned be limited to 2? It should not be 3 if I've changed the limit to be 2.